src/git/git.go

```diff
@@ -53,6 +72,345 @@ func (g *GitVersioning) validate() error {
 	return nil
 }
 
+func (g *GitVersioning) GetChangeHash() (string, error) {
+	if g.mostRecentCommit.hash == "" {
+		return "", errors.New("last commit hash not set")
+	}
+	return g.mostRecentCommit.hash, nil
+}
+
+func (g *GitVersioning) GetChangeAuthorName() (string, error) {
+	if g.mostRecentCommit.authorName == "" {
+		return "", errors.New("last commit name not set")
+	}
+	return g.mostRecentCommit.authorName, nil
+}
+
+func (g *GitVersioning) GetChangeAuthorEmail() (string, error) {
+	if g.mostRecentCommit.authorEmail == "" {
+		return "", errors.New("last commit email not set")
+	}
+	return g.mostRecentCommit.authorEmail, nil
+}
+
+func (g *GitVersioning) GetChangeMessage() (string, error) {
+	fmt.Println(g.mostRecentCommit.message)
+	if g.mostRecentCommit.message == "" {
+		return "", errors.New("last commit message not set")
+	}
+	return g.mostRecentCommit.message, nil
+}
+
+func (g *GitVersioning) GetCurrentVersion() (string, error) {
+	if g.mostRecentTag == "" {
+		return "", errors.New("most recent tag not set")
+	}
+	return g.mostRecentTag, nil
+}
+
+func (g *GitVersioning) UpgradeRemoteRepository(newVersion string) error {
+	// consider 1.0.0 as the start tag of a repository when it does not have tags yet
+	if newVersion == "0.1.0" || newVersion == "0.0.1" {
+		newVersion = "1.0.0"
+	}
+
+	if err := g.commitChanges(newVersion); err != nil {
+		return err
+	}
+
+	if err := g.push(); err != nil {
+		return err
+	}
+
+	if err := g.setTag(newVersion); err != nil {
+		return err
+	}
+
+	if err := g.pushTags(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// getBranchPointedToHead aims to get the branch reference pointing to HEAD.
+// Returns:
+// 		error: Error whenever something wrong happen.
+func (g *GitVersioning) getBranchPointedToHead() error {
+	defer g.printElapsedTime("GetBranchPointedToHead")()
+	g.log.Info("getting branch pointed to HEAD")
+	ref, err := g.repo.Head()
+	if err != nil {
+		return fmt.Errorf("error while retrieving the branch pointed to HEAD due to: %s", err.Error())
+	}
+
+	g.branchHead = ref
+
+	return nil
+}
+
+// getCommitHistory aims to retrieve the commit history from a given repository.
+// Returns:
+// 		error: Error whenever something wrong happen.
+func (g *GitVersioning) getCommitHistory() error {
+	defer g.printElapsedTime("GetComitHistory")()
+	g.log.Info("getting commit history")
+	cIter, err := g.repo.Log(&git.LogOptions{From: g.branchHead.Hash(), Order: git.LogOrderCommitterTime})
+	if err != nil {
+		return fmt.Errorf("error while retrieving the commit history  due to: %s", err.Error())
```

Comment:
what do you think use operator `w` ?

https://go.dev/blog/go1.13-errors

@esequielvirtuoso

Reply:
I've changed it.